dashboard/app/models/user.rb

```diff
@@ -2261,6 +2261,41 @@ def providers
     end
   end
 
+  # Returns number of times a user has attempted to join a section in the last 24 hours
+  # Returns nil if no section join attempts
+  def num_section_attempts
+    properties['section_attempts'] || 0
```

Comment:
What is `properties` supposed to be used for? Why is the code using `properties` instead of adding a new attribute to the `User` model?

Reply:
Most new fields of models are put into `properties`. The fields in `properties` of the User model are [here](https://github.com/code-dot-org/code-dot-org/blob/0cfe36263ee85b4581622fe928ce3c2f413b347e/dashboard/app/models/user.rb#L96). I think `section_attempts` makes sense in `properties`, unless there's a reason to query for it directly.

Reply:
But on that note, `section_attempts` and `section_attempts_last_reset` should be added to that last I linked

Reply:
Which also should allow you to reference them directly instead of through `properties`, which might've been what Dayne was going for all along....

(Sorry for multiple comments, it's still early so my brain is still booting)

Reply:
@bethanyaconnor I didn't know what you were talking about so thank you for the explanation. I didn't see any other direct uses of the 'properties' field and that field has no documentation, so it seemed weird to me. Is this a common active record pattern? Does it have a name I could look up and learn more about it?

Reply:
@daynew Yeah sorry for the extra comments 😬  I believe this is a custom thing defined [here](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/concerns/serialized_properties.rb).

This is definitely one of those things that I didn't know existed until I had to add a new attribute last summer

Reply:
wow, didn't know about this either. thanks all for enlightening me --- this will definitely clean up the code